### PR TITLE
feat: initial implementation of mem to ref_gl

### PR DESCRIPTION
**Pull request by pollend**

### CODE REVIEW

File: source/qcommon/mod_mem.h [hunk 1]
```diff
@@ -6,47 +6,88 @@
...
+	Mem_ValidationAllAllocations, \
+	Q_Free, \
+	Q_CreatePool, \
+	Q_LinkToPool, \
+	Q_FreePool, \
+	Q_EmptyPool \
+};
+
 #if MEM_DEFINE_INTERFACE_IMPL
 static struct mem_import_s mem_import;
```
Comment (learn-more) [inline source/qcommon/mod_mem.h:74]: Every file including this header will get it's own copy of this struct, that is probably not what you want.

File: source/client/cl_vid.c [hunk 2]
```diff
@@ -316,7 +317,11 @@ static bool VID_LoadRefresh( const char *name )
 	GetRefAPI_t GetRefAPI_f;
 
 	VID_UnloadRefresh();
+	
+	vid_ref_mempool = Q_CreatePool( NULL, "Refresh" );
+	struct mem_import_s memImport = DECLARE_MEM_STRUCT(vid_ref_mempool); 
 
+	import.memImport = &memImport;
```
Comment (learn-more) [inline source/client/cl_vid.c:324]: This becomes a dangling reference as soon as `VID_LoadRefresh` returns, so you might as well skip the dance of storing it in a local variable and returning a pointer to that, and just store the entire thing directly in `import` ?

Comment (pollend) [reply source/client/cl_vid.c:324]: C doesn't have a dangling reference its a point to the struct that declares the import. I guess the second the function returns I the pointer become undefined. its only declared for this function so its probably not too big of an issue?

File: source/qcommon/mod_mem.h (hunk 1 above)
Comment (pollend) [reply source/qcommon/mod_mem.h:74]: I addressed this comment.

File: source/client/cl_vid.c (hunk 2 above)
Comment (pollend) [reply source/client/cl_vid.c:328]: I added an assertion and made the variable static so there is a dropped pointer

**Pull request merged**